test-basic/concurrent-requests.js

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2020 MarkLogic Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+var testconfig = require('../etc/test-config.js');
+
+var marklogic = require('../');
+
+var dbWriter = marklogic.createDatabaseClient(testconfig.restWriterConnection);
+
+describe('concurrent-requests test', function(){
+
+
+    it('should write 1000 documents concurrently', function(done){
+        for(var i=0; i<3; i++) {
+            writeBatch(i,0);
+        }
+
+        done();
+    });
+});
+
+
+function writeBatch(writerId, nextDocument) {
+    if(nextDocument>=1000) {
+        console.log(writerId + ' is done');
```

Comment:
I'd suggest calling done() here when all of the writes have been queued.  That's sufficient for the prototype.

In the real version, the code will need to notify the caller only when the writes have finished.